datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -1013,20 +1038,27 @@ async fn concatenate_parallel_row_groups(
     )?;
 
     while let Some(task) = serialize_rx.recv().await {
+        let mut rg_reservation =
+            MemoryConsumer::new("ParquetSink(SerializedRowGroupWriter)").register(&pool);
```

Comment:
Same question: does the name used have any uniqueness significance?
If yes, then we could use the bytes_written offset (prior to start writing this row group).